packages/react/src/QuestionnaireForm/QuestionnaireFormItem/QuestionnaireFormItem.tsx

```diff
@@ -429,3 +457,45 @@ function typedValueToString(typedValue: TypedValue | undefined): string | undefi
   }
   return typedValue.value.toString();
 }
+
+function getItemsByLinkId(allResponses: QuestionnaireResponseItem[], linkId: string): QuestionnaireResponseItem[] {
+  let result: QuestionnaireResponseItem[] = [];
+
+  for (const item of allResponses) {
+    // If the linkId matches, add it to the result array
+    if (item.linkId === linkId) {
+      result.push(item);
+    }
+
+    // If the current item has nested items, search them too
+    if (item.item) {
+      result = result.concat(getItemsByLinkId(item.item, linkId));
+    }
+  }
+  return result;
+}
+
+function getItemValue(answer: QuestionnaireResponseItemAnswer): any {
+  const itemValue = getTypedPropertyValue(
+    { type: 'QuestionnaireItemAnswerOption', value: answer },
+    'value'
+  ) as TypedValue;
+  // formatCoding returns '' if nothing is there so we need to use ||
+  return formatCoding(itemValue?.value) || itemValue?.value;
```

Comment:
Looked into CodingDisplay and used this, figure this is more elegant than an if statement but if you prefer that or a ternary operator I'll change it